File: 백준/Silver/11497. 통나무 건너뛰기/통나무 건너뛰기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {

        StringBuilder sb = new StringBuilder();

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());

        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine());

            int[] arr = new int[n];
            int[] temp = new int[n];

            StringTokenizer st = new StringTokenizer(br.readLine());

            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(st.nextToken());
            }

            Arrays.sort(arr);

            int left = 0, right = n - 1;
            for (int i = 0; i < n; i++) {
                if (i % 2 == 0) {
                    temp[left++] = arr[i];
                } else {
                    temp[right--] = arr[i];
                }
            }

            int ans = Math.abs(temp[n - 1] - temp[0]);
            for (int i = 0; i < n - 1; i++) {
                ans = Math.max(ans, Math.abs(temp[i + 1] - temp[i]));
            }

            sb.append(ans).append("\n");
        }

        System.out.print(sb);
    }
}

File: 백준/Silver/11497. 통나무 건너뛰기/통나무 건너뛰기.py
t = int(input())

for _ in range(t):

    n = int(input())
    arr = list(map(int, input().split()))
    temp = [0] * n

    arr.sort()

    left, right = 0, n - 1

    for i in range(n):
        if i % 2 == 0:
            temp[left] = arr[i]
            left += 1
        else:
            temp[right] = arr[i]
            right -= 1

    ans = abs(temp[n - 1] - temp[0])

    for i in range(n - 1):
        ans = max(ans, abs(temp[i + 1] - temp[i]))

    print(ans)